extension/src/frequency.rs

```diff
@@ -399,6 +399,92 @@ pub mod toolkit_experimental {
     }
 
     ron_inout_funcs!(SpaceSavingAggregate);
+
+    pg_type! {
+        #[derive(Debug)]
+        struct SpaceSavingBigIntAggregate<'input> {
+            num_values: u32,
+            topn: u32,
+            values_seen: u64,
+            freq_param: f64,
+            counts: [u64; self.num_values], // JOSH TODO look at AoS instead of SoA at some point
+            overcounts: [u64; self.num_values],
+            datums: [i64; self.num_values],
+        }
+    }
+
+    impl<'input> From<&SpaceSavingTransState> for SpaceSavingBigIntAggregate<'input> {
+        fn from(trans: &SpaceSavingTransState) -> Self {
+            assert_eq!(trans.type_oid(), pg_sys::INT8OID);
+
+            let mut values = Vec::new();
+            let mut counts = Vec::new();
+            let mut overcounts = Vec::new();
+
+            for entry in &trans.entries {
```

Comment:
I demonstrate how to genericize this in commit 31e12c3e9f38f07b9fce3ea8279ae5b4a1674ad6